runtime/vm/jvminit.c

```diff
@@ -4056,6 +4056,24 @@ processVMArgsFromFirstToLast(J9JavaVM * vm)
 		vm->checkpointState.javaDebugThreadCount = 0;
 	}
 
+#if defined(LINUX)
+	{
+		IDATA enableCRIUDisclaimClasses = FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXENABLECRIUDISCLAIMCLASSES, NULL);
+		IDATA disableCRIUDisclaimClasses = FIND_AND_CONSUME_VMARG(EXACT_MATCH, VMOPT_XXDISABLECRIUDISCLAIMCLASSES, NULL);
+		if (enableCRIUDisclaimClasses > disableCRIUDisclaimClasses) {
+			PORT_ACCESS_FROM_JAVAVM(vm);
+			UDATA *pageSizes = j9vmem_supported_page_sizes();
+			/* Only disclaim classes for 4K page sizes. */
+			if (4096 == pageSizes[0]) {
```

Comment:
Please note for Linux PPC LE small page size `pageSizes[0]` is 64k, I believe the logic you are adding is applicable too. Also I am not sure what you are checking here. If class segments always use small pages check is pointless. If you trying to check are segments allocated with Huge Pages this check should be different (check page size in memory identifier returned by Port during memory reservation)

Reply:
@tajila
- Should classes be disclaimed with `madvise` only for 4KB pages or for all pages below 4KB?
- Are we avoiding pages larger than 4KB because disclaim might not be effective ... touching a page will make a large amount of memory to become resident?

Reply:
>     Are we avoiding pages larger than 4KB because disclaim might not be effective ... touching a page will make a large amount of memory to become resident?

Correct, disclaim is not as effective with larger page sizes.